tests/utils/test_annotations.py

```diff
@@ -105,7 +157,7 @@ def test_multi_line_docstring_second_line():
 
 
 def test_deprecated_and_keyword_first():
-    docstring = deprecated_and_keyword_only_first.__doc__
+    docstring = str(deprecated_and_keyword_only_first.__doc__)
```

Comment:
Is `__doc__` not a string?

Reply:
It's Optional[str]

Reply:
why do we need `str` here?

Reply:
The pylance warnings annoy me. It's immaterial and I've reverted them since the docstrings can't be empty here

Reply:
what does pylance complain here?

Reply:
pylance complains about typing correctness. Since a __doc__ value isn't known if it is supplied with any entry (depending on if a docstring is set or not) until runtime evaluation, it simply warns that the type definition is invalid. 

It's not a warning I like to turn off in VSCode as many times it can point out some typing issues in otherwise reasonable situations. 

```
Operator "in" not supported for types "Literal['``tests.utils.test_annotations.AnotherDeprecatedDa…']" and "str | None"
```

Reply:
what happen if you insert `assert __doc__ is not None`?

Reply:
`assert __doc__ is not None` can silence the warning:

https://github.com/user-attachments/assets/0bed8683-436a-428e-acf8-df07cbfd02e2

This conveys the intent more clearly.

Reply:
yep and definitely makes the assertions more meaningful. Great idea!